middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -463,11 +463,9 @@ let close_c_call acc env ~loc ~let_bound_ids_with_kinds
   in
   let kind_of_primitive_native_repr
       ((_, repr) : Primitive.mode * Primitive.native_repr) =
-    (* CR layouts v2: This match will be extended with [| Same_as_ocaml_repr
-       Float64 -> K.naked_float] in the PR that adds Float64. *)
     match repr with
-    | Same_as_ocaml_repr Value -> K.value
-    | Same_as_ocaml_repr Void -> assert false
+    | Same_as_ocaml_repr sort ->
+      K.With_subkind.(kind (from_lambda (Typeopt.layout_of_const_sort sort)))
```

Comment:
We can certainly just directly translate the sort here.  (In a previous iteration you were concerned about having too many independent translations of sorts, which is how I ended up here, but I do probably prefer the old version on balance.  Perhaps there is some middle-ground, but I don't see it at the moment)

Reply:
Yeah, I know I'm pulling in both ways here (and I guessed you were trying to respond to my other comments). I still wonder if simply defining several functions all in one place where they can be checked for cross-consistency (even if only by hand) would be better. But this might be a good thing to brainstorm with e.g. Mark at the meeting tomorrow.

Reply:
I wouldn't worry about this - presumably flambda2 should be able to optimize this, and the amount of allocation will be minimal anyway.